extensions/amp-story-embed/0.1/amp-story-embed.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpStoryEmbedManager} from '../../../src/amp-story-embed-manager';
+import {Layout} from '../../../src/layout';
+
+export class AmpStoryEmbed extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    this.embedManager_ = new AmpStoryEmbedManager(
```

Comment:
I think it is safer to have components do 0 work in the constructor. Is there any downside to moving this to `buildCallback`?
If for some reason we need to instantiate the `AmpStoryEmbedManager` before the buildCallback, maybe we could use `firstAttachedCallback`

Reply:
Sounds good, I will change that. Just wondering: why do you think this is safer?

Reply:
Doing some work in the constructor of an element will run it as soon as the element is attached to the DOM, very likely during rendering.
Because the work is "hidden" in the other constructor of the EmbedManager, it would be easy to introduce latency or privacy issues by mistake, just because it is hard to realize when the code actually runs if you don't have context on the whole thing.